Content.Server/Codewords/CodewordSystem.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Linq;
+using Content.Server.Administration.Logs;
+using Content.Server.GameTicking.Rules;
+using Content.Shared.Database;
+using Content.Shared.GameTicking.Components;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Server.Codewords;
+
+/// <summary>
+/// Gamerule that provides codewords for other gamerules that rely on them.
+/// </summary>
+public sealed class CodewordSystem : GameRuleSystem<CodewordRuleComponent>
+{
+    [ValidatePrototypeId<EntityPrototype>]
+    public static string RuleComponent = "CodewordRule";
```

Comment:
we typically do a `private static EntProtoId` instead of the validator attribute. Also, if you do that, you can use the templated version to validate that the entity prototype has a certain component (`CodewordComponent`)